src/RestApi/StoreApi/Controllers/CartShippingRates.php

```diff
@@ -128,6 +135,32 @@ function( $item ) {
 		return rest_ensure_response( $response );
 	}
 
+	/**
+	 * Update the selected shipping method.
+	 *
+	 * @param \WP_Rest_Request $request Full data about the request.
+	 * @return \WP_Error|\WP_REST_Response Response object on success, or WP_Error object on failure.
+	 */
+	public function update_item( $request ) {
+		$controller = new CartController();
+		$cart       = $controller->get_cart_instance();
+
+		if ( ! $cart || ! $cart instanceof \WC_Cart ) {
+			return new RestError( 'woocommerce_rest_cart_error', __( 'Unable to retrieve cart.', 'woo-gutenberg-products-block' ), array( 'status' => 500 ) );
+		}
+
+		$chosen_shipping_methods = WC()->session->get( 'chosen_shipping_methods' );
+		$posted_shipping_methods = isset( $request['shipping_method'] ) ? wc_clean( wp_unslash( $request['shipping_method'] ) ) : array();
```

Comment:
I'm not sure we can do it this way because you'd be trusting the data from the client which could mean shipping amounts could be adjusted and then stored in session?

Reply:
I tried to use the same method used in wc ajax actions, this sends the method id only, and the order creation should use that.